src/Data/Text.hs

```diff
@@ -664,7 +608,7 @@ compareLength t n = S.compareLengthI (stream t) n
 -- >>> T.map (\c -> if c == '.' then '!' else c) message
 -- "I am not angry! Not at all!"
 --
--- Subject to fusion.  Performs replacement on invalid scalar values.
+-- Performs replacement on invalid scalar values.
```

Comment:
Aren't `map` and other functions involving `stream` or `unstream` in their definition still subject to fusion? the `stream/unstream` rule in `Data.Text.Internal.Fusion` is still there.

Reply:
See above in https://github.com/haskell/text/pull/348#issue-668392402:
> some function may still happen to fuse by chance (if they have no materialized implementation at all), but this is no longer promised

`stream . unstream = id` on its own is a reasonable contraction, which is always beneficial, so I did not have an immediate justification to scrap it. This means that some combinations, e. g., `length . map` can still accidentally fuse, because `length = S.length . stream` and `map f = unstream . S.map f . stream` unconditionally. However, I removed the promise of fusion from haddocks even for `length` and `map`, so that we are free to roll out non-streaming implementations in future. For example, I have in mind a faster SIMD-based implementation of `length`, which does not waste time on decoding/encoding of a stream and operate on raw bytes instead.